src/org/pcic/util/Utils.groovy

```diff
@@ -0,0 +1,35 @@
+package org.pcic.util
+
+import org.pcic.GitUtils
+import org.pcic.DockerUtils
+
+
+class Utils implements Serializable {
+    def steps
+
+    Utils(steps) {
+        this.steps = steps
+    }
+
+    Map processParams(Map expected, Map args) {
+        def unknownKeys = (expected.keySet() + args.keySet()) - expected.keySet()
+
+        if (unknownKeys.size() > 0) {
+            throw new Exception("Key(s) ${unknownKeys} not available in ${expected.keySet()}")
+        }
+
+        return expected + args
```

Comment:
Doing a little judicious guessing here (instead of looking up the language ref), it looks like what this function does is update `expected` with `args`, and throw an error if `args` contains any key not in `expected`. Yes? In any case, a comment explaining what it computes would be helpful. And maybe a more explanatory name than `processParams` would be helpful; something like `argsWithDefaults` perhaps.